Common/cpp/SharedItems/Shareables.h

```diff
@@ -0,0 +1,448 @@
+#pragma once
+
+#include <jsi/jsi.h>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef RCT_NEW_ARCH_ENABLED
+#include <react/renderer/uimanager/primitives.h>
+#endif
+
+#include "ReanimatedRuntime.h"
+#include "RuntimeManager.h"
+#include "Scheduler.h"
+
+#define HAS_JS_WEAK_OBJECTS JS_RUNTIME_HERMES
```

Comment:
Does V8 runtime also support `jsi::WeakObject`? cc @Kudo

Reply:
yes, it did: https://github.com/Kudo/react-native-v8/blob/96a1a2c8a35349967a3ad7219106a3475b85433a/src/v8runtime/V8Runtime.cpp#L982-L997